kedro/framework/context/context.py

```diff
@@ -216,8 +200,12 @@ def params(self) -> dict[str, Any]:
         except MissingConfigException as exc:
             warn(f"Parameters not found in your Kedro project config.\n{str(exc)}")
             params = {}
-        _update_nested_dict(params, self._extra_params or {})
-        return params
+
+        if self._extra_params:
+            # Merge nested structures
+            params = OmegaConf.merge(params, self._extra_params)
```

Comment:
Would this accidentally open up using omegaconf feature instead of just pure dictionary merge?